libs/tokens/src/pure/TokenLogo/index.tsx

```diff
@@ -160,3 +119,90 @@ export function TokenLogo({
     </Styled.TokenLogoWrapper>
   )
 }
+
+type LpTokenLogoProps = Omit<TokenLogoProps, 'token'> & { token: LpToken }
+
+function LpTokenLogo({ token, className, size = 36, sizeMobile }: LpTokenLogoProps): ReactNode {
+  const tokensByAddress = useTokensByAddressMap()
+
+  return (
+    <Styled.TokenLogoWrapper className={className} size={size} sizeMobile={sizeMobile}>
+      <Styled.LpTokenWrapper size={size}>
+        <div>
+          <TokenLogo noWrap token={tokensByAddress[token.tokens?.[0]]} size={size} sizeMobile={sizeMobile} />
+        </div>
+        <div>
+          <TokenLogo noWrap token={tokensByAddress[token.tokens?.[1]]} size={size} sizeMobile={sizeMobile} />
+        </div>
+      </Styled.LpTokenWrapper>
+    </Styled.TokenLogoWrapper>
+  )
+}
+
+interface TokenLogoUrlOptions {
+  token?: TokenWithLogo | Currency | null
+  logoURI?: string
+  invalidUrls: Record<string, boolean>
+}
+
+function useTokenLogoUrl({ token, logoURI, invalidUrls }: TokenLogoUrlOptions): {
+  currentUrl?: string
+  initial?: string
+} {
+  const urls = useMemo(() => {
+    if (token instanceof LpToken) {
+      return []
+    }
+
+    if (token instanceof NativeCurrency) {
+      return [cowprotocolTokenLogoUrl(NATIVE_CURRENCY_ADDRESS.toLowerCase(), token.chainId as SupportedChainId)]
+    }
+
+    if (token) {
+      return getTokenLogoUrls(token as TokenWithLogo)
+    }
+
+    return logoURI ? uriToHttp(logoURI) : []
```

Comment:
It's better to define a constant EMPTY_ARRAY = [] and reuse it here to prevent rerenders

Reply:
https://github.com/cowprotocol/cowswap/pull/6518#discussion_r2541623814